index.js

```diff
@@ -443,58 +445,84 @@ class HtmlWebpackInlineSVGPlugin {
 
-        return new Promise((resolve, reject) => {
+        const config = {}
 
-            const svgSrc = this.getImagesSrc(inlineImage)
 
+        // pass all objects to the config.plugins array
 
-            // if the image isn't valid resolve
+        config.plugins = _.map(configObj, (value, key) => ({ [key]: value }));
 
-            if (!svgSrc) return resolve(html)
 
+        // create a new instance of SVGO
+        // passing it the merged config, to optimize the svg
 
-            // read in the svg
+        const svgo = new SVGO(config)
 
-            fs.readFile(path.resolve(this.outputPath, svgSrc), 'utf8', (err, data) => {
-
-                if (err) reject(err)
+        svgo.optimize(data)
+            .then((result) => {
 
-                const configObj = Object.assign(svgoDefaultConfig, this.userConfig)
+                const optimisedSVG = result.data
 
-                const config = {}
+                html = this.replaceImageWithSVG(html, inlineImage, optimisedSVG)
 
+                resolve(html)
 
-                // pass all objects to the config.plugins array
+            })
+            .catch((err) => console.log(chalk.red(err)))
+    }
 
-                config.plugins = _.map(configObj, (value, key) => ({ [key]: value }));
 
+    /**
+     * append the inlineImages SVG data to the output HTML and remove the original img by
+     * loading the SVG data from the filesystem or from an URL
+     * @param {string} html
+     * @param {Object} inlineImage - parse5 document
+     * @returns {Promise}
+     *
+     */
+    processOutputHtml (html, inlineImage) {
+        return new Promise((resolve, reject) => {
 
-                // create a new instance of SVGO
-                // passing it the merged config, to optimize the svg
+            const svgSrc = this.getImagesSrc(inlineImage)
 
-                const svgo = new SVGO(config)
+            // if the image isn't valid resolve
+            if (!svgSrc) return resolve(html)
 
-                svgo.optimize(data)
-                    .then((result) => {
+            // read in the svg
+            fs.readFile(path.resolve(this.outputPath, svgSrc), 'utf8', (err, data) => {
+                if (err) {
+                    // loading from the filesystem failed
+                    if (!this.allowFromUrl) {
+                        return reject(err)
+                    }
 
-                        const optimisedSVG = result.data
+                    axios.get(svgSrc)
+                        .then(({ status, data }) => {
+                            if (!status === 200) {
+                                resolve(html)
+                                return
+                            }
 
-                        html = this.replaceImageWithSVG(html, inlineImage, optimisedSVG)
+                            this.optimizeSvg({ html, inlineImage, svgSrc, data, resolve })
+                        })
+                        .catch(() => resolve(html))
```

Comment:
Same. Shouldn't this be `.catch(error => reject(error))`?

Reply:
I don't think so, the flow looks like this (Promise-based):
- `updateHTML` -> `inlineImages.reduce()`
- `processImage(html)
- `processOutputHtml(html, inlineImage)

The intention is to update the HTML on each iteration, by selecting the image intended to be replaced within the process.

If one of the images isn't available, at that iteration the nested Promise can be resolved with the current state of the html. 

Only if the image is available via filesystem or URL, the `optimizeSvg` gets called and therefore the `replaceImageWithSVG`, also resolving the promise.

If we're rejecting the promise then the HTML does not get correctly updated.

[This comment also applies to your other comment](https://github.com/theGC/html-webpack-inline-svg-plugin/pull/29#discussion_r440359958).

Reply:
So take the filesystem example as a base:
- If the file does not get found or can't be read, it's fair to reject the promise and miserably fail the process.
- If the file exists, it gets replaced with the content and the promise gets resolved so it can iterate to the next image.

Why such process? Because there is the expectancy of needing existing Svg files in the first place at the project, and that totally makes sense.

When loading files through network I think the plugin should be resilient to not fail in case the URL does not resolve correctly, meaning:
- If the URL resolves to a proper content, replace it (HTTP status 200 and with content).
- If the URL does not resolve correctly, for whatever reason, simply not replacing the image would be enough.

Reply:
[See here](https://github.com/theGC/html-webpack-inline-svg-plugin/pull/29#discussion_r440651980).